elastic/kibana
# [Index Management] Add index mode field in index template form

Closes https://github.com/elastic/kibana/issues/198620

## Summary

This PR adds a field for index mode setting in the Logistics step in Index Template form.


https://github.com/user-attachments/assets/ee38bdec-66ff-468d-a55e-abf5354c3da2


**How to test:**
1. Go to Index Management -> Index Templates and start creating an index template
2. Verify that the index mode is only enabled if the data stream toggle is on.
3. Verify that typing the `logs-*-*` index pattern sets the index mode to "LogsDB":

<img width="1401" alt="Screenshot 2024-11-13 at 13 00 10" src="https://github.com/user-attachments/assets/d1825d08-5039-4c43-80a8-653233e0b677">



4. Go to the Settings step and verify that the index mode callout is displayed correctly.
5. Go to Review step and verify that Index mode is displayed correctly in both the summary and the preview request.
6. Save the template and verify that the template details tab correctly displays the index mode.

<img width="1565" alt="Screenshot 2024-11-13 at 17 22 54" src="https://github.com/user-attachments/assets/2055501b-32c9-463c-b61d-541b9687b459">

<img width="1565" alt="Screenshot 2024-11-13 at 17 22 31" src="https://github.com/user-attachments/assets/21c9cf9e-5858-4403-9106-57ed8ccf3639">





### Checklist

Delete any items that are not applicable to this PR.
